challenge_9/python/shaowen/square_sort.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Jan 10 08:25:26 2017
+
+@author: Shaowen Liu
+"""
+
+def square_sort(arr):
+    # retieve the negtive and non-negtive list,O(n)
+    a_list = [-i for i in arr if i < 0]
+    b_list = arr[len(a_list):][::-1]
```

Comment:
just let you know, 
arr[:len(a_list)-1:-1] is sure efficient, but in my origin code, len(a_list) could be zero, in this case it will be b_list = arr[:-1:-1],  =>> b_list = [ ]. 
This's why, in the new commits, I treat the case (len(a_list) == 0) separately.